app/models/contact_info.rb

```diff
@@ -12,7 +12,7 @@ class ContactInfo < ActiveRecord::Base
   validates :type, presence: true, inclusion: { in: VALID_TYPES }
   validates :value,
             presence: true,
-            uniqueness: {scope: [:type]}
+            uniqueness: {scope: [:user_id, :type]}
```

Comment:
I don't think they should.  We have implicit assumptions that when we find matching emails during SessionsCallback that refers to the same user.  I'll ask the group at large, copying you.